pkg/storage/common.go

```diff
@@ -68,6 +71,36 @@ func ValidateManifest(imgStore ImageStore, repo, reference, mediaType string, bo
 	switch mediaType {
 	case ispec.MediaTypeImageManifest:
 		var manifest ispec.Manifest
+		sc := jsonschema.Reflect(&manifest)
+		b, _ := json.Marshal(sc)
+		loader := gojsonschema.NewStringLoader(string(b))
+		documentLoader := gojsonschema.NewStringLoader(string(body))
+		schema, err := gojsonschema.NewSchema(loader)
+		if err != nil {
+			log.Error().Err(err).Msg("unable to unmarshal JSON")
+
+			return "", zerr.ErrBadManifest
+		}
+
+		result, err := schema.Validate(documentLoader)
```

Comment:
Is it possible to initialize the `schema` beforehand, when the server starts? Instead of creating these objects every time.

Reply:
Will have to become a global var likely :(